### DISCUSSION TITLE
Unable to install appium plugins. Can any one help please

### DISCUSSION THREAD

**Discussion by lambodar19** [appium/appium#19506] [General]
I am trying to install the appium plugins but getting the below error. I have tried to upgrade/downgrade appium. I have version 2.0.0 now but somewhere its showing conflict with 1.22.3 which I am unable to find 

✖ Installing 'appium-device-farm'
Error: ✖ Encountered an error when installing package: npm command 'install --save-dev appium-device-farm --json' failed with code 1.
STDOUT:
{
  "error": {
    "code": "ERESOLVE",
    "summary": "unable to resolve dependency tree",
    "detail": "\nWhile resolving: undefined@undefined\nFound: appium@1.22.3\nnode_modules/appium\n  appium@\"^1.22.3\" from the root project\n\nCould not resolve dependency:\npeer appium@\"^2.2.1\" from appium-device-farm@8.4.0\nnode_modules/appium-device-farm\n  dev appium-device-farm@\"*\" from the root project\n\nFix the upstream dependency conflict, or retry\nthis command with --force or --legacy-peer-deps\nto accept an incorrect (and potentially broken) dependency resolution."
  }
}

STDERR:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: undefined@undefined
npm ERR! Found: appium@1.22.3
npm ERR! node_modules/appium
npm ERR!   appium@"^1.22.3" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer appium@"^2.2.1" from appium-device-farm@8.4.0
npm ERR! node_modules/appium-device-farm
npm ERR!   dev appium-device-farm@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:

**Here is the log file**

'Log files:
C:\Users\91900\AppData\Local\npm-cache\_logs\2023-12-09T14_40_50_540Z-debug-0.log

# npm resolution error report

While resolving: undefined@undefined
Found: appium@1.22.3
node_modules/appium
  appium@"^1.22.3" from the root project

Could not resolve dependency:
peer appium@"^2.2.1" from appium-device-farm@8.4.0
node_modules/appium-device-farm
  dev appium-device-farm@"*" from the root project

Fix the upstream dependency conflict, or retry
this command with --force or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.

**Comment by lambodar19** [comment]
here is the log file- # npm resolution error report

While resolving: undefined@undefined
Found: appium@1.22.3
node_modules/appium
  appium@"^1.22.3" from the root project

Could not resolve dependency:
peer appium@"^2.2.1" from appium-device-farm@8.4.0
node_modules/appium-device-farm
  dev appium-device-farm@"*" from the root project

Fix the upstream dependency conflict, or retry
this command with --force or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.

**Reply by KazuCocoa** [reply]
Please uninstall appium 1.22.3 from your project or machine. It looks like appum 1.22.3 remained in your machine somewhere.

**Reply by lambodar19** [reply]
Thanks for the response @KazuCocoa I have already uninstalled it. Please see the CLI below. I manually deleted the old appium files also but unable to locate the version 1.22.3. Is there anyway ? C:\Users\91900>npm list -g appium
C:\Program Files\nodejs -> .\
`-- appium@2.0.0
  `-- @appium/docutils@0.4.13
    `-- @appium/typedoc-plugin-appium@0.6.6
      `-- appium@2.0.0 deduped

**Reply by KazuCocoa** [reply]
I'm not sure, but for example, we previously observed user-specific env remained 1.22.3.
https://github.com/appium/appium/discussions/17109

One possible reason is nodejs was installed by multiple methods such as homebrew and nvm etc, and some of them keep its `node_modules` in their place, but others do not. It is basically nodejs environment stuff.

It would be nice to check all possible `node_modules` directories on your environment.

**Reply by lambodar19** [reply]
Thanks a lot @KazuCocoa . Issue got resolved. When I upgraded appium multiple node_modules folder got created which was causing the issues. I removed everything and installed fresh it worked.
